src/MainWindow/main_window.cpp

```diff
@@ -1220,6 +1271,24 @@ bool MainWindow::confirmExitProgram() {
               QMessageBox::No | QMessageBox::Yes) == QMessageBox::Yes);
 }
 
+void MainWindow::setVisibleRefreshButtons(bool visible) {
+  for (auto dock : m_pinAssignmentDocks) {
+    auto button = dock->findChild<QPushButton*>();
```

Comment:
dock->findChild<QPushButton*> will return whatever first button it finds which is probably ok in a limited scenario where we know the widget construction, but this could have unexpected results down the road. Might want to set the objectName of your refresh button then update findChild to `findChild<QPushButton*>("refreshBtnName")`

Reply:
It makes sense, updated!